pkg/drivers/libhvee/libhvee_windows.go

```diff
@@ -0,0 +1,309 @@
+		return ErrPowerShellNotFound
+	}
+
+	// Check that hyperv is installed
+	if err := hypervAvailable(); err != nil {
+		return err
+	}
+
+	// Check that the user is an Administrator
+	isAdmin, err := isAdministrator()
+	if err != nil {
+		return err
+	}
+	if !isAdmin {
+		return ErrNotAdministrator
+	}
+
+	return nil
+}
+
+func (d *Driver) Create() error {
+	log.Debugf("Machine: libhvee -> creating: system settings")
+	systemSettings, err := hypervctl.NewSystemSettingsBuilder().
+		PrepareSystemSettings(d.MachineName, nil).
+		PrepareMemorySettings(
+			func(ms *hypervctl.MemorySettings) {
+				ms.DynamicMemoryEnabled = d.DynamicMemory
+				ms.VirtualQuantity = uint64(d.Memory)
+				ms.Reservation = 1024
+				ms.Limit = uint64(d.Memory)
+			}).
+		PrepareProcessorSettings(
+			func(ps *hypervctl.ProcessorSettings) {
+				ps.VirtualQuantity = uint64(d.CPU)
+			}).
+		Build()
+
+	if err != nil {
+		return err
+	}
+
+	log.Debugf("Machine: libhvee -> creating: copy disk image")
+	diskPath := d.getDiskPath()
+	if err := crcos.CopyFile(d.ImageSourcePath, diskPath); err != nil {
+		return err
+	}
+
+	log.Debugf("Machine: libhvee -> creating: hardware setup")
+	err = hypervctl.NewDriveSettingsBuilder(systemSettings).
+		AddScsiController().
+		AddSyntheticDiskDrive(0).
+		DefineVirtualHardDisk(diskPath,
+			func(vhdss *hypervctl.VirtualHardDiskStorageSettings) {}).
+		Finish().
+		Finish().
+		Finish().
+		Complete()
+
+	if err != nil {
+		return err
+	}
+
+	log.Debugf("Machine: libhvee -> creating: done")
+
+	return nil
+}
+
+func (d *Driver) getMachine() (*hypervctl.VirtualMachine, error) {
+	vmm := hypervctl.NewVirtualMachineManager()
+
+	log.Debugf("Machine: libhvee -> get machine")
+	return vmm.GetMachine(d.MachineName)
+}
+
+// Start starts an host
+func (d *Driver) Start() error {
+	vm, err := d.getMachine()
+	if err != nil {
+		return err
+	}
+
+	log.Debugf("Machine: libhvee -> start")
+	return vm.Start()
+}
+
+// Stop stops an host
+func (d *Driver) Stop() error {
+	vm, err := d.getMachine()
+	if err != nil {
+		return err
+	}
+
+	log.Debugf("Machine: libhvee -> stop")
+	return vm.Stop()
+}
+
+// Remove removes an host
+func (d *Driver) Remove() error {
+	s, err := d.GetState()
+	if err != nil {
+		return err
+	}
+
+	log.Debugf("Machine: libhvee -> remove running")
+	if s == state.Running {
+		if err := d.Kill(); err != nil {
+			return err
+		}
+	}
+
+	vm, err := d.getMachine()
+	if err != nil {
+		return err
+	}
+
+	log.Debugf("Machine: libhvee -> remove vm")
+	err = vm.Remove("")
+	if err != nil {
+		return err
+	}
+
+	log.Debugf("Machine: libhvee -> remove disk file")
+	return os.Remove(d.getDiskPath())
+}
+
+// Kill force stops an host
+func (d *Driver) Kill() error {
+	vm, err := d.getMachine()
+	if err != nil {
+		return err
+	}
+
+	log.Debugf("Machine: libhvee -> stop with force")
+	return vm.StopWithForce()
+}
+
+func (d *Driver) GetIP() (string, error) {
+	return "", fmt.Errorf("")
```

Comment:
Can this return a better error? This could be `return "", drivers.ErrNotSupported`
(`drivers` is 	`"github.com/crc-org/machine/libmachine/drivers"`)